docs/data/data-grid/selection/CellSelectionGrid.tsx

```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+import Button from '@mui/material/Button';
+import { DataGridPremium } from '@mui/x-data-grid-premium';
+import { useDemoData } from '@mui/x-data-grid-generator';
+
+export default function CellSelectionGrid() {
+  const [rowSelection, setRowSelection] = React.useState(false);
+
+  const { data } = useDemoData({
+    dataSet: 'Commodity',
+    rowLength: 10,
+    maxColumns: 6,
+  });
+
```

Comment:
Is it possible to scroll automatically when selecting cells range?
Currently, using a mouse you can only select cell range within visible cells


https://user-images.githubusercontent.com/13808724/200921355-9957b82a-47f0-450f-baf6-78b8b2cb2a0a.mov

Reply:
In terms of UX, from the sources that I could benchmark:

- Airtable: OK but it feels that it starts scrolling too far away from the bottom of the viewport. It's distracting.
- Google Sheet: OK but it feels that it starts scrolling too late, only one you are beyond the bottom of the viewport. You don't see what you are going to select next.
- Notion: Pretty strong, the threshold feels good, and they also have an incremental scrolling speed based on far you are from the bottom of the viewport 👌.

Reply:
> It's similar to what we have for the column headers (although it doesn't work).

Yes, there's an open issue for it https://github.com/mui/mui-x/issues/6236